pkg/apis/core/v1/defaults.go

```diff
@@ -349,6 +349,28 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+	// TODO, remove this in 1.22
+	// we cant SetDefaults for nameless namespaces (generateName),
+	// we don't need to overwrite the pre-existing labelMetadataName
+	if len(obj.Name) > 0 && obj.Labels[v1.LabelMetadataName] != obj.Name {
+		if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
+		}
```

Comment:
@jayunit100 static check is complaining about this empty if

Maybe following @liggitt here would be better, and dropping everything else :)

```go
if len(obj.Name) > 0 && obj.Labels[v1.LabelMetadataName] != obj.Name {
  if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
     	if obj.Labels == nil {
		obj.Labels = map[string]string{}
	}
        obj.Labels[v1.LabelMetadataName] = obj.Name
  }
}
```